firmware/cmake/embedded.cmake

```diff
@@ -0,0 +1,285 @@
+include("${CMAKE_SOURCE_DIR}/firmware/cmake/shared.cmake")
```

Comment:
I would argue that if you want shared code, just separately import it to prevent double running the code? There's only one other spot I see you using embedded without shared, and it seems like it shouldn't be too much head bandwidth to import both shared and embedded, especially since most of the meat is in shared, and embedded if you need embedded stuff